spring-ai-core/src/main/java/org/springframework/ai/chat/client/advisor/RetrievalAugmentationAdvisor.java

```diff
@@ -119,30 +127,45 @@ private AdvisedRequest before(AdvisedRequest request) {
 		Map<String, Object> context = new HashMap<>(request.adviseContext());
 
 		// 0. Create a query from the user text and parameters.
-		Query query = new Query(new PromptTemplate(request.userText(), request.userParams()).render());
+		Query originalQuery = new Query(new PromptTemplate(request.userText(), request.userParams()).render());
+
+		// 1. Transform original user query based on a chain of query transformers.
+		Query transformedQuery = originalQuery;
+		for (var queryTransformer : queryTransformers) {
+			transformedQuery = queryTransformer.apply(transformedQuery);
+		}
 
-		// 1. Retrieve similar documents for the original query.
-		List<Document> documents = this.documentRetriever.retrieve(query);
+		// 2. Retrieve similar documents for the original query.
+		List<Document> documents = this.documentRetriever.retrieve(transformedQuery);
 		context.put(DOCUMENT_CONTEXT, documents);
```

Comment:
maybe a RAGContext that in turn contains the documents so that for the future when additional things are added to the context for RAG, it is all in a central spot?

Reply:
We agreed to talk more about this when we reconsider the design of ChatResponse and how to propagate back evidence from different intermediate steps in a ChatClient, such as retrieved documents and called functions.